app/api/src/Security/Voter/UserVoter.php

```diff
@@ -10,22 +11,29 @@
 class UserVoter extends Voter
 {
     private $security = null;
+    private $kernel = null;
 
     /**
      * @param Security $security
+     * @param KernelInterface $kernel
      */
-    public function __construct(Security $security)
+    public function __construct(Security $security, KernelInterface $kernel)
     {
         $this->security = $security;
+        $this->kernel = $kernel;
     }
     protected function supports(string $attribute, mixed $subject): bool
     {
-        return in_array($attribute, ['USER_READ', 'USER_EDIT', 'USER_DELETE'])
-            && $subject instanceof \App\Entity\User;
+        return (in_array($attribute, ['USER_READ', 'USER_EDIT', 'USER_DELETE'])
+            && $subject instanceof \App\Entity\User) || $this->kernel->getEnvironment() === 'test';
```

Comment:
instead
$this->kernel->getEnvironment() === 'test'
consider
'test' === $this->kernel->getEnvironment()